File: Front end/src/app/models/vehicle.ts
import { VehicleType } from "./vehicle-type";

export class Vehicle {
    id: string;
    name: string;
    typeID: string;
    vehicleType: VehicleType;

    constructor(id: string, name: string, typeID: string, vehicleType: VehicleType) {
        if (id == null && vehicleType == null) {
            this.name = name;
            this.typeID = typeID;
        }
        else {
            this.id = id;
            this.name = name;
            this.typeID = typeID;
            this.vehicleType = new VehicleType(vehicleType.id, vehicleType.name);
        }
    }
}

File: VehicleRental/VehicleRental.Models/Vehicle.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace VehicleRental.Models
{
    public class Vehicle
    {
        public Guid ID { get; set; }

        public string Name { get; set; }

        public Guid TypeID { get; set; }

        [ForeignKey("TypeID")]
        public virtual VehicleType VehicleType { get; set; }
    }
}
